plugins/quickhighlight/pluma-quickhighlight-plugin.c

```diff
@@ -0,0 +1,482 @@
+    GtkSourceSearchSettings *search_settings;
+    GtkTextIter start, end;
+    g_autofree gchar *text = NULL;
+
+    g_assert (PLUMA_IS_QUICK_HIGHLIGHT_PLUGIN (plugin));
+
+    plugin->priv->queued_highlight = 0;
+
+    if (!gtk_text_buffer_get_selection_bounds (GTK_TEXT_BUFFER (plugin->priv->buffer), &start, &end))
+    {
+        g_clear_object (&plugin->priv->search_context);
+        return G_SOURCE_REMOVE;
+    }
+
+    if (gtk_text_iter_get_line (&start) != gtk_text_iter_get_line (&end))
+    {
+        g_clear_object (&plugin->priv->search_context);
+        return G_SOURCE_REMOVE;
+    }
+
+    if (plugin->priv->search_context == NULL)
+    {
+        search_settings =
+            g_object_new (GTK_SOURCE_TYPE_SEARCH_SETTINGS,
+                          "at-word-boundaries", FALSE,
+                          "case-sensitive", TRUE,
+                          "regex-enabled", FALSE,
+                          NULL);
+
+        plugin->priv->search_context =
+            g_object_new (GTK_SOURCE_TYPE_SEARCH_CONTEXT,
+                          "buffer", plugin->priv->buffer,
+                          "highlight", FALSE,
+                          "match-style", plugin->priv->style,
+                          "settings", search_settings,
+                          NULL);
+
+        g_object_unref (search_settings);
+    }
+    else
+    {
+        search_settings =
+            gtk_source_search_context_get_settings (plugin->priv->search_context);
+    }
+
+    text = gtk_text_iter_get_slice (&start, &end);
+
+    gtk_source_search_settings_set_search_text (search_settings, text);
+
+    gtk_source_search_context_set_highlight (plugin->priv->search_context, TRUE);
+
+    return G_SOURCE_REMOVE;
+}
+
+static void
+pluma_quick_highlight_plugin_queue_update (PlumaQuickHighlightPlugin *plugin)
+{
+    g_return_if_fail (PLUMA_IS_QUICK_HIGHLIGHT_PLUGIN (plugin));
+
+    if (plugin->priv->queued_highlight != 0)
+    {
+        return;
+    }
+
+    plugin->priv->queued_highlight =
+        gdk_threads_add_idle_full (G_PRIORITY_LOW,
+                                   pluma_quick_highlight_plugin_highlight_worker,
+                                   g_object_ref (plugin),
+                                   g_object_unref);
+}
+
+static void
+pluma_quick_highlight_plugin_notify_weak_buffer_cb (gpointer data,
+                                                    GObject *where_the_object_was)
+{
+    PlumaQuickHighlightPlugin *plugin = PLUMA_QUICK_HIGHLIGHT_PLUGIN (data);
+
+    g_assert (PLUMA_IS_QUICK_HIGHLIGHT_PLUGIN (plugin));
+
+    plugin->priv->style_scheme_handler_id = 0;
+    plugin->priv->buffer = NULL;
+}
+
+static void
+pluma_quick_highlight_plugin_unref_weak_buffer (PlumaQuickHighlightPlugin *plugin)
+{
+    g_return_if_fail (PLUMA_IS_QUICK_HIGHLIGHT_PLUGIN (plugin));
+
+    if (plugin->priv->buffer == NULL)
+    {
+        return;
+    }
+
+    if (plugin->priv->delete_range_handler_id > 0)
```

Comment:
`plugin->priv->delete_range_handler_id` is an unsigned long integer, so you must check that the value is different from zero, check the other identifiers (_handler_id).

Reply:
It checks whether the value is greater than zero or what do you mean?

Reply:
An unsigned long cannot be lower than zero.
```C
if (handler_id != 0)
  {
    g_signal_handler_disconnect (instance, handler_id);
    handler_id = 0;
  }
```
or
```C
if (handler_id)
  {
    g_signal_handler_disconnect (instance, handler_id);
    handler_id = 0;
  }
```
or
```C
#if GLIB_CHECK_VERSION(2,62,0)
g_clear_signal_handler (&handler_id, instance);
#else
if (handler_id != 0)
  {
    g_signal_handler_disconnect (instance, handler_id);
    handler_id = 0;
  }
#endif
```
or
```C
#if GLIB_CHECK_VERSION(2,62,0)
g_clear_signal_handler (&handler_id, instance);
#else
if (handler_id)
  {
    g_signal_handler_disconnect (instance, handler_id);
    handler_id = 0;
  }
#endif
```